src/main/java/io/lettuce/core/resource/EventLoopResources.java

```diff
@@ -58,6 +60,11 @@
      */
     Class<? extends Channel> socketChannelClass();
 
+    /**
+     * @return the {@link DatagramChannel} class.
+     */
+    Class<? extends DatagramChannel> datagramChannelClass();
```

Comment:
[Question; nits]
For consistency with `socketChannelClass()`, is it better to declare here as `Class<? extends Channel> datagramChannelClass()`?
It seems a bit wired because a method for TCP returns `Channel` while a method for UDP returns `DatagramChannel`.
Note that if we return `Class<? extends Channel> ` here, we need cast the return value at
https://github.com/lettuce-io/lettuce-core/pull/1517/files#diff-954f0da1e02d2af777f9afc758b8b20f3f2df40a221f1a0e3569ef05402ccd97R310